### DISCUSSION TITLE
Hot water temp reading

### DISCUSSION THREAD

**Discussion by ahti87** [klejejs/python-thermia-online-api#46] [Q&A]
Is it possible to read hot water temperature as well in addition to all other metrics?

**Comment by wouterse** [comment]
I have been using heat_pump.get_register_data_by_register_group_and_name to obtain other metrics besides the ones that are available as heatpump.xx, such as heat_pump.heat_temperature.

As an example I was interested in the value of integral. In the example.py file I simply added a line at the end to obtain this value (before adding it to my own code):

print ("integral: " + str(heat_pump.get_register_data_by_register_group_and_name(
            	"REG_GROUP_OPERATIONAL_STATUS", "REG_INTEGRAL_LSD")["value"]))

All you need to know is the name of the register_group and the register_name. I used the files debug.py (that is produced by example.py) to find these two names. I looked for 'hot_water' in the debug files and found "REG_GROUP_TEMPERATURES" and "REG_HOT_WATER_TEMPERATURE". Then I tried:

print ("hot water temerature " + str(heat_pump.get_register_data_by_register_group_and_name(
            	"REG_GROUP_TEMPERATURES", "REG_HOT_WATER_TEMPERATURE")["value"]))

And indeed that produces the correct value for me.

**Reply by fclauson** [reply]
@wouterse could you send me an example of how you used this in HA 
I want to do just that (read a value)
and then ideally I want to write one back to the HP
